### PR TITLE
Temp table cleanup

### PR DESCRIPTION
**Pull request by davidrabinowitz**

### CODE REVIEW

File: bigquery-connector-common/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryClient.java [hunk 1]
```diff
@@ -810,8 +828,10 @@ TableInfo createTableFromQuery() {
       if (job.getStatus().getError() != null) {
         throw BigQueryUtil.convertToBigQueryException(job.getStatus().getError());
       }
+      // Registering a cleanup job
+      CLEANUP_JOBS.add(() -> bigQueryClient.deleteTable(tempTable));
       // add expiration time to the table
-      TableInfo createdTable = bigQueryClient.getTable(destinationTable);
+      TableInfo createdTable = bigQueryClient.getTable(tempTable);
       long expirationTime =
           createdTable.getCreationTime() + TimeUnit.MINUTES.toMillis(expirationTimeInMinutes);
```
Comment (vishalkarve15) [inline bigquery-connector-common/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryClient.java:836]: Can we avoid this change by setting the expiration time during creation?

**Review by vishalkarve15 — APPROVED** [review]
Approved with a question.

File: bigquery-connector-common/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryClient.java (hunk 1 above)
Comment (davidrabinowitz) [reply bigquery-connector-common/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryClient.java:836]: Unfortunately no....

**Pull request merged**